### ISSUE TITLE
GasNET segsize issue

### ISSUE THREAD

**Issue by bandokihiro** [StanfordLegion/legion#860]
When running my code with the following command
```
mpiexec -np 2 -ppn 1 solver -logfile logs/log_%.log -hdf5:forcerw -ll:cpu 2 -ll:rsize 10000 -ll:csize 5000
```
I get the following GasNET error
```
*** WARNING (proc 0): GASNet gasnetc_attach returning an error code: GASNET_ERR_BAD_ARG (Invalid function parameter passed)
  at /nobackupp2/kbando/MyLibsSource/gasnet/GASNet-2019.9.0/ibv-conduit/gasnet_core.c:2865
  reason: segsize too large
GASNET: gasnet_attach(handlers, hcount, attach_size, 0) = 10003 (GASNET_ERR_BAD_ARG, Invalid function parameter passed)
*** WARNING (proc 1): GASNet gasnetc_attach returning an error code: GASNET_ERR_BAD_ARG (Invalid function parameter passed)
  at /nobackupp2/kbando/MyLibsSource/gasnet/GASNet-2019.9.0/ibv-conduit/gasnet_core.c:2865
  reason: segsize too large
GASNET: gasnet_attach(handlers, hcount, attach_size, 0) = 10003 (GASNET_ERR_BAD_ARG, Invalid function parameter passed)
```
I have also set those 2 envionment variables
```
export GASNET_PHYSMEM_MAX=20G
export GASNET_MAX_SEGSIZE=20G
```
I have 64GB per node and I am using the configuration from the Legion repository with ibv conduit. The problem does not show up if I set `-ll:rsize 9000`.

**Comment by streichler** [comment]
How are you configuring gasnet?  Our canned config (https://github.com/StanfordLegion/gasnet/blob/master/configs/config.ibv.release) sets a limit of 16GB at compile time.  Still, that should be well north of the 10G you're asking for.
As a quick sanity-check, can you run with `-level activemsg=2`?  That'll display exactly what realm is asking gasnet for.

**Comment by bandokihiro** [comment]
I am using that canned configuration. My understanding was that the environment variable overrides the compile-time limit.

**Comment by bandokihiro** [comment]
```
[0 - 2aaaaab035c0]    0.000000 {2}{activemsg}: Pinned Memory Usage: GASNET=0 MB, RMEM=-2032 MB, IBRMEM=0 MB, LMB=4 MB, SDP=64 MB, total=17592186042452 MB
```

**Comment by streichler** [comment]
Yeah, that's bad.  As a side note, I think `-rsize 3000` would have failed also.  And although `-rsize 9000` appears to work, it's not actually allocating 9GB.  The problem is 32-bit integer overflow - I should have a fix shortly.

**Comment by bandokihiro** [comment]
This problem is solved with commit dd6993621ea15e06856c709b4e6de396a9122a8f
